dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -1389,4 +1368,111 @@ private void CaptureUsageDetails(CompletionsUsage usage)
             await functionCallCallback(context).ConfigureAwait(false);
         }
     }
+
+    /// <summary>
+    /// Configures the function calling functionality based on the provided parameters.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> to be used for function calling.</param>
+    /// <param name="executionSettings">Execution settings for the completion API.</param>
+    /// <param name="chatOptions">The chat completion options from the Azure.AI.OpenAI package.</param>
+    /// <param name="requestIndex">Request sequence index of automatic function invocation process.</param>
+    /// <returns>A tuple containing the AllowAnyRequestedKernelFunction and MaximumAutoInvokeAttempts settings.</returns>
+    private (bool? AllowAnyRequestedKernelFunction, int? MaximumAutoInvokeAttempts)? ConfigureFunctionCalling(Kernel? kernel, OpenAIPromptExecutionSettings executionSettings, ChatCompletionsOptions chatOptions, int requestIndex)
+    {
+        if (executionSettings.FunctionChoiceBehavior is not null && executionSettings.ToolCallBehavior is not null)
+        {
+            throw new ArgumentException("ToolBehaviors and ToolCallBehavior cannot be used together.");
```

Comment:
> I also think that we should not throw an exception and use configuration based on priority. If both are specified, use FunctionChoiceBehavior as a new approach, which will overwrite ToolCallBehavior, we can mention this behavior in XML documentation.

Both solutions look good to me. The one that throws the exception notifies the caller immediately when the code is invoked, potentially saving time they would spend troubleshooting unexpected behavior. The one that relies on priority is more permissive and allows callers to use both solutions simultaneously, even at the cost of potential confusion or unexpected behavior. I would probably go with the proposed "priority" one taking into account its permissiveness. Let's see what others think.

> New approach also allows to specify configuration in json/yaml. So, if I have already existing working code with ToolCallBehavior, and I want to try new configuration in json/yaml, I will receive an exception at first try, because it will force me to remove ToolCallBehavior from my existing code. Choosing one property based on priority will require minimal changes from developers.

Today, as far as I understand, prompt execution settings provided via the code override the ones that come from prompts, so no exception will be thrown in this scenario. To allow execution settings specified in the prompt to be applied, either the settings specified in the code should be removed or a different service selector should be used.

Reply:
We plan to remove the `ToolCallBehavior` so the functionality will be dropped as well.